pkg/services/cosmosdb/common_update.go

```diff
@@ -1,11 +1,47 @@
 package cosmosdb
 
 import (
+	"context"
 	"fmt"
+	"reflect"
 
 	"github.com/Azure/open-service-broker-azure/pkg/service"
 )
 
+func (
+	c *cosmosAccountManager,
+) ValidateUpdatingParameters(instance service.Instance) error {
+	pp := instance.ProvisioningParameters
+	up := instance.UpdatingParameters
+
+	if err := validateReadLocations(
+		"graph account update",
+		up.GetStringArray("readRegions"),
+	); err != nil {
+		return err
+	}
+
+	// Can't update readRegions and other properties at the same time
+	ppData := make(map[string]interface{})
+	upData := make(map[string]interface{})
+	for k, v := range pp.Data {
+		ppData[k] = v
+	}
+	for k, v := range up.Data {
+		upData[k] = v
+	}
+	ppData["readRegions"] = nil
+	upData["readRegions"] = nil
```

Comment:
Using built-in function delete(upData, "readRegions") is better.

Reply:
Thanks, will change it.